test/any_type.test.js

```diff
@@ -76,7 +76,7 @@ describe('any type', function () {
     });
 
     assert(fn.signatures instanceof Object);
-    assert.deepEqual(Object.keys(fn.signatures), ['string,any', 'any']);
+    assert.deepEqual(Object.keys(fn.signatures), ['any', 'string,any']);
     assert.equal(fn('foo', 2), 'string,any');
```

Comment:
Note that the changes in ordering signatures did affect this function. But the order of these two particular signatures makes no difference because neither has a rest parameter and they have a different number of parameters, so they can _never_ both match anyway. The proposed code puts shorter non-rest signatures earlier to try to take as much advantage as possible of the implementation's short-circuiting of the first six signatures as long as they have two or fewer arguments.